server/src/main/java/org/elasticsearch/cluster/coordination/StableMasterHealthIndicatorService.java

```diff
@@ -0,0 +1,208 @@
+                    }
+                    stableMasterStatus = HealthStatus.YELLOW;
+                    summary = String.format(
+                        Locale.ROOT,
+                        "The cluster's master has alternated between %s and no master multiple times in the last 30 minutes",
+                        master
+                    );
+                    impacts.add(
+                        new HealthIndicatorImpact(
+                            3,
+                            "The cluster is at risk of not being able to create, delete, or rebalance indices",
+                            List.of(ImpactArea.INGEST)
+                        )
+                    );
+                    if (includeDetails) {
+                        details.put("current_master", new DiscoveryNodeXContentObject(localMasterHistory.getCurrentMaster()));
+                    }
+                } else {
+                    logger.trace(
+                        String.format(
+                            Locale.ROOT,
+                            "This node thinks the master is unstable, but the master node %s thinks it is stable",
+                            master
+                        )
+                    );
+                    stableMasterStatus = HealthStatus.GREEN;
+                    summary = "The cluster has a stable master node";
+                }
+            } else if (masterChanges > 3) {
+                logger.trace("Have seen {} master changes in the last 30 minutes", masterChanges);
+                stableMasterStatus = HealthStatus.YELLOW;
+                summary = String.format(Locale.ROOT, "The master has changed %d times in the last 30 minutes", masterChanges);
+                impacts.add(
+                    new HealthIndicatorImpact(
+                        3,
+                        "The cluster currently has a master node, but having multiple master node changes in a short time is an indicator "
+                            + "that the cluster is at risk of of not being able to create, delete, or rebalance indices",
+                        List.of(ImpactArea.INGEST)
+                    )
+                );
+                if (includeDetails) {
+                    List<DiscoveryNode> mastersInLast30Minutes = localMasterHistory.getImmutableView();
+                    details.put("current_master", new DiscoveryNodeXContentObject(localMasterHistory.getCurrentMaster()));
+                    details.put("recent_masters", mastersInLast30Minutes.stream().map(DiscoveryNodeXContentObject::new).toList());
+                }
+            } else {
+                logger.trace("The cluster has a stable master node");
+                stableMasterStatus = HealthStatus.GREEN;
+                summary = "The cluster has a stable master node";
+            }
+        } else {
+            stableMasterStatus = HealthStatus.GREEN;
```

Comment:
I had set it to GREEN to make it abundantly clear that this path had not been implemented. :) I'll change it to RED in case someone winds up using this.